src/Simulation/Simulators.Tests/QubitManagerTests.cs

```diff
@@ -152,6 +152,91 @@ public void TestQubitManager()
             }
         }
 
+        /// <summary>
+        /// Test for QubitManager.
+        /// </summary>
+        [Fact]
+        public void TestQubitManagerDiscouragingReuse()
```

Comment:
Split into two tests so it will be easier to localize test failures?

Reply:
Yes, the same allocation pattern is being tested, just in another function. I am trying to avoid having too many test functions. We already have about 300 and navigating that list is a concern. So I am trying to keep small related tests together. This way it is also easier to understand they are doing the same. See for yourself, you did notice the same pattern in two blocks because they were in the same function, but you didn't notice the same pattern in a function right above it.

Reply:
I didn't look at the function above, because that code didn't change -- that's an inherent danger of code reviews. If I was working with code (to change it, etc.) I'd look through test cases' names and this is where having well named separate tests helps to build the mental picture of what is covered and what matters. The other benefit of smaller tests is that when a test fails you get a localized failure, identified by the test's name. Might not be a big deal on a local machine when you can rerun tests under debugger, but important for CIs.